text/0000-layer-metadata-files.md

```diff
@@ -0,0 +1,184 @@
+# Meta
+[meta]: #meta
+- Name: Combine and organize metadata file locations across lifecycle
+- Start Date: 2021-03-11
+- Author(s): [jabrown85](https://github.com/jabrown85) [samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This is a proposal to change the default locations of several files and directories. Part of the reasoning behind this is to disambiguate metadata files from application or layer files and avoid naming collisions.
+
+# Motivation
+[motivation]: #motivation
+
+Currently all the metadata files live under the `<layers>` folder. This means that application metadata files like `launch.toml`, `build.toml` or `store.toml` share the same directory as layer metadata files. Since the layers could possibly be named `launch`, `build` or `store` this leads to ambiguity, conflicts and/or restrictions on layer names and their associated metadata files that should not exist. This also prevents us from possibly adding new metadata files in the future without breaking backwards compatibility with buildpacks who may have been creating layers with the same name. This proposal seeks to disambiguate special metadata files from free-form layer metadata files.
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposal is as follows -
+
+The current folder structure is -
+
+```
+/cnb (provided by builder image)
+├── buildpacks
+│   └── <buildpacks>
+├── lifecycle
+│   └── <creator/builder/etc>
+├── order.toml
+└── stack.toml
+/layers (configurable with `CNB_LAYERS_DIR` or -layers)
+├── group.toml
+├── plan.toml
+├── order.toml (written by a platform, optionally)
+├── report.toml
+├── <escaped_buildpack_id> (buildpack arg `$1`)
+│   ├── build.toml
+│   ├── launch.toml
+│   ├── store.toml
+│   ├── <layer>.toml
+│   └── <layer>
+└── config
+    └── metadata.toml
+/workspace (configurable with `CNB_APP_DIR ` or -app)
+└── <app source code to transform>
+```
+
+This proposal is to change this folder structure to -
+
+```
+/cnb (provided by builder image)
+├── buildpacks
+│   └── <buildpacks>
+├── lifecycle
+│   └── <creator/builder/etc>
+├── order.toml
+└── stack.toml
+/workspace (configurable with `CNB_WORKSPACE_DIR`)
+├── app (configurable with `CNB_APP_DIR ` or -app)
+│    └── <app source code to transform>
+├── platform-config (configurable with `CNB_PLATFORM_CONFIG_DIR`)
+│   └── order.toml (written by a platform, optionally - prior to `lifecycle` execution `platform/prepare`)
+├── lifecycle-config (configurable with `CNB_LIFECYCLE_CONFIG_DIR`)
+│   └── group.toml
+│   └── plan.toml
+│   └── report.toml
```

Comment:
This isn't really a "config" file - is there another place for it?

Reply:
Yeah, this whole directory is more like `lifecycle-output`. Now that I think about it...`analyzed.toml` is missing from the example and needs a new home. I assumed it would go here, but the name `config` is maybe not the best given the contents.

Reply:
I think that `project-metadata.toml` is also missing.

Reply:
Added